website/payments/forms.py

```diff
@@ -29,6 +29,19 @@ class Meta:
         model = BankAccount
 
 
+class BankAccountUserRevokeForm(forms.ModelForm):
+    """
+    Custom form for members to revoke their bank account
+    """
+
+    def is_valid(self):
+        return self.instance.can_be_revoked
```

Comment:
I believe you are bypassing eventual other checks here by not calling super.

Reply:
You are absolutely right